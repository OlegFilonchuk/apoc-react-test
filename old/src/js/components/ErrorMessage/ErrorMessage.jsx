import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

import './ErrorMessage.less';

function ErrorMessage({ isVisible, children, className }) {
  if (isVisible) {
    const finalClassName = classNames('error-message', className);

    return (
      <div className={finalClassName}>
        <span className="error-message--icon">
          <i className="sc-icon-triangle-warning" />
        </span>
        <span className="error-message--text">{children}</span>
      </div>
    );
  }

  return null;
}

ErrorMessage.propTypes = {
  /**
   * Whether alert should be opened or closed by default
   */
  isVisible: PropTypes.bool,

  /**
   * Can be either simple text or link
   */
  children: PropTypes.node.isRequired,

  /**
   * Any custom class names
   */
  className: PropTypes.string
};

ErrorMessage.defaultProps = {
  isVisible: false,
  children: '',
  className: ''
};

export default ErrorMessage;
