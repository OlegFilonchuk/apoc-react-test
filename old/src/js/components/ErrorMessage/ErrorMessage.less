@error-message-color: #da1134;
@sign-color: #fff;

.error-message {
  display: flex;
  flex-direction: row;
  justify-content: flex-start;

  padding-top: 5px;

  color: @error-message-color;

  &--icon {
    font-size: 16px;
    margin-right: 5px;
  }

  &--text {
    padding-top: 5px;
  }
}
