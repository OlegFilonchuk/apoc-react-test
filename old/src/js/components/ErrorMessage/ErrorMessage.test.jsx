import React from 'react';
import { mount } from 'enzyme';

import ErrorMessage from './ErrorMessage';

const mountAlert = (content, props = {}, mountMethod = mount) =>
  mountMethod(<ErrorMessage {...props}>{content}</ErrorMessage>);

describe('<ErrorMessage>', () => {
  it('should fire onClose event when clicking on close button', () => {
    const content = 'Test Message';
    const errorMessage = mountAlert(content, mount);

    expect(errorMessage).toMatchSnapshot();
  });
});
