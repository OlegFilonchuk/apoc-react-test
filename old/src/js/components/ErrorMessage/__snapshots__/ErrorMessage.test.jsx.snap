// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorMessage> should fire onClose event when clicking on close button 1`] = `
<ErrorMessage
  className=""
  isVisible={false}
/>
`;
